File: concrete/src/main/java/pcb/uwu/unit/derived/termodynamics/PascalUnit.java
package pcb.uwu.unit.derived.termodynamics;

import pcb.uwu.units.derived.termodynamics.PressureUnit;

import static pcb.uwu.unit.base.KiloGramUnit.KILOGRAM;
import static pcb.uwu.unit.base.MeterUnit.METER;
import static pcb.uwu.unit.base.SecondUnit.SECOND;

public class PascalUnit extends PressureUnit {

	public static final PascalUnit PASCAL = new PascalUnit();

	public PascalUnit() {
		super(METER, KILOGRAM, SECOND);
	}

	@Override
	public String getSymbol() {
		return "Pa";
	}

	@Override
	public String getSingularName() {
		return "pascal";
	}

	@Override
	public String getPluralName() {
		return "pascals";
	}
}

File: implementation/src/main/java/pcb/uwu/unit/derived/termodynamics/PascalUnit.kt
package pcb.uwu.unit.derived.termodynamics

import pcb.uwu.unit.base.KiloGramUnit
import pcb.uwu.unit.base.MeterUnit
import pcb.uwu.unit.base.SecondUnit

object PascalUnit
    : PressureUnit(lengthUnit = MeterUnit,
                   massUnit = KiloGramUnit,
                   timeUnit = SecondUnit)
{
    override val symbol = "Pa"
    override val singularName = "pascal"
    override val pluralName = "pascals"
}
